langlib/lang.string/src/main/ballerina/string.bal

```diff
@@ -309,27 +432,32 @@ public isolated function padZero(string str, int len, Char zeroChar = "0") retur
     name: "padZero"
 } external;
 
-# Refers to the `RegExp` type defined by lang.regexp module.
-public type RegExp regexp:RegExp;
-
 # Tests whether there is a full match of a regular expression with a string.
 # A match of a regular expression in a string is a full match if it
 # starts at index 0 and ends at index `n`, where `n` is the length of the string.
 # This is equivalent to `regex:isFullMatch(re, str)`.
 #
+# ```ballerina
+# "Ballerina".matches(re `Bal.*`) ⇒ true
+# ```
+#
 # + str - the string
 # + re - the regular expression
 # + return - true if there is full match of `re` with `str`, and false otherwise
-public function matches(string str, RegExp re) returns boolean {
-   return re.isFullMatch(str);
+public isolated function matches(string str, RegExp 're) returns boolean {
+   return 're.isFullMatch(str);
 }
 
 # Tests whether there is a match of a regular expression somewhere within a string.
 # This is equivalent to `regexp:find(re, str, startIndex) != ()`.
 #
+# ```ballerina
+# "New Zealand".includesMatch(re `lan?`) ⇒ true
+# ```
+#
 # + str - the string to be matched
 # + re - the regular expression
 # + return - true if the is a match of `re` somewhere within `str`, otherwise false
-public function includesMatch(string str, RegExp re, int startIndex = 0) returns boolean {
-   return re.find(str, startIndex) != ();
+public isolated function includesMatch(string str, RegExp 're, int startIndex = 0) returns boolean {
+   return 're.find(str, startIndex) != ();
```

Comment:
@pcnfernando Are there any difference between the these regex functions in master and 3.x ?

Reply:
Updated as per the offline chat.